### DISCUSSION TITLE
Map status checking

### DISCUSSION THREAD

**Discussion by banderovskyi** [giorgiabosello/google-maps-react-markers#213] [Help] [answered]
I could use some help dealing with the 'For development purposes only' watermark on the map.
Is there a way to detect when this message shows up proactively? If so, I'd like to set up a log file or some other solution to skip the hassle of manually checking the map.

**Comment by giorgiabosello** [comment]
@banderovskyi It appears when the apiKey is not set or it is wrong or the URL of the website where you are using the map is not enabled in the cloud console. 
So, before trying to implement something weird, watch for the apiKey!

**Reply by banderovskyi** [reply]
Yep, you're right. But in my case, clients can set their API keys and I don't have access there.
But I need to know when it happens before users see it.
That is the reason why I'm asking.

**Reply by giorgiabosello** [reply, marked as answer]
@banderovskyi the only thing you can do is checking if the apiKey is set before rendering the map. The map layer is automatically set by Google, there's no control over it. 
You could check the script url too, apiKey is there also, but I suggest you to do the other way.

**Reply by banderovskyi** [reply]
Okay thanks
